src/rust/ide/src/controller/searcher.rs

```diff
@@ -578,57 +587,150 @@ impl Searcher {
         executor::global::spawn(async move {
             let this_type = this_type.await;
-            info!(logger,"Requesting new suggestion list. Type of `this` is {this_type:?}.");
-            let request  = ls.completion(&file,&position,&this_type,&return_type,&tags);
-            let response = request.await;
-            info!(logger,"Received suggestions from Language Server.");
-            let new_suggestions = match response {
-                Ok(list) => {
-                    let entry_ids   = list.results.into_iter();
-                    let entries     = entry_ids.filter_map(|id| {
-                        let entry = database.get(id);
-                        if entry.is_none() {
-                            error!(logger,"Missing entry {id} in Suggestion Database.");
-                        }
-                        entry
-                    });
-                    let suggestions = entries.map(Suggestion::Completion);
-                    Suggestions::Loaded {list:Rc::new(suggestions.collect())}
-                },
+            info!(this.logger,"Requesting new suggestion list. Type of `this` is {this_type:?}.");
+            let requests  = return_types.into_iter().map(|return_type| {
+                info!(this.logger, "Requesting suggestions for returnType {return_type:?}");
+                let file = graph.module.path().file_path();
+                ls.completion(file,&position,&this_type,&return_type,&tags)
+            });
+            let responses = futures::future::join_all(requests).await;
+            info!(this.logger,"Received suggestions from Language Server.");
+            let new_suggestions = match this.suggestions_from_responses(responses) {
+                Ok(list)   => Suggestions::Loaded {list:Rc::new(list)},
                 Err(error) => Suggestions::Error(Rc::new(error.into()))
             };
-            data.borrow_mut().suggestions = new_suggestions;
-            notifier.publish(Notification::NewSuggestionList).await;
+            this.data.borrow_mut().suggestions = new_suggestions;
+            this.notifier.publish(Notification::NewSuggestionList).await;
         });
-        Ok(())
+    }
+
+    /// Process multiple completion responses from the engine into a single list of suggestion.
+    fn suggestions_from_responses
+    (&self, responses:Vec<json_rpc::Result<language_server::response::Completion>>)
+    -> FallibleResult<Vec<Suggestion>> {
+        let mut suggestions = Vec::new();
+        for response in responses {
+            let response = response?;
+            let entries  = response.results.iter().filter_map(|id| {
+                self.database.get(*id).map_none(||
+                    error!(self.logger,"Missing entry {id} in Suggestion Database.")
+                ).map(Suggestion::Completion)
+            });
+            suggestions.extend(entries);
+        }
+        Ok(suggestions)
+    }
+
+    fn possible_function_calls(&self) -> Vec<CompletionSuggestion> {
```

Comment:
We could also use information about function arity to reject some overloads (e.g. if entry takes 2 parameters and we already provided 3 arguments).

Reply:
Those are filtered out anyway in `return_types_for_argument_completion`